mockup/patterns/relateditems/pattern.js

```diff
@@ -286,12 +293,13 @@ define([
 
           // Filter out items:
           // While browsing: always include folderish items
-          // Browsing and searching: Only include selectable items, which are not already selected.
+          // Browsing and searching: Only include selectable items which are not already selected, and all folders
+          // even if they're selected, as we need them available for browsing/selecting their children
           results = results.filter(
             function (item) {
               if (
                 (this.browsing && item.is_folderish) ||
-                (this.isSelectable(item) && this.selectedUIDs.indexOf(item.UID) == -1)
+                (this.isSelectable(item) && !this.selectedUIDs.includes(item.UID))
```

Comment:
oops, `includes` doesn't work for IE11 without a polyfill. I'll check if we have on in place otherwise I'll revert these changes.